driver/src/main/scala/com/stratio/sparkta/driver/actor/FragmentSupervisorActor.scala

```diff
@@ -0,0 +1,121 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.stratio.sparkta.driver.actor
+
+import akka.actor.Actor
+import akka.event.slf4j.SLF4JLogging
+import com.stratio.sparkta.driver.dto.FragmentElementDto
+import com.stratio.sparkta.sdk.JsoneyStringSerializer
+import org.apache.curator.framework.CuratorFramework
+import org.json4s.DefaultFormats
+import org.json4s.ext.EnumNameSerializer
+import org.json4s.native.Serialization._
+import spray.httpx.Json4sJacksonSupport
+
+import scala.collection.JavaConversions
+
+/**
+ * List of all possible akka messages used to manage fragments.
+ */
+case class FragmentSupervisorActor_create(fragment: FragmentElementDto)
+case class FragmentSupervisorActor_findAllByType(fragmentType: String)
+case class FragmentSupervisorActor_findByTypeAndName(fragmentType: String, name: String)
+case class FragmentSupervisorActor_deleteByTypeAndName(fragmentType: String, name: String)
+case class FragmentSupervisorActor_response_error(exception: Exception)
+case class FragmentSupervisorActor_response_fragment(fragment: FragmentElementDto)
+case class FragmentSupervisorActor_response_fragments(fragments: Seq[FragmentElementDto])
+case class FragmentSupervisorActor_response_ok()
+
+/**
+ * Implementation of supported CRUD operations over ZK needed to manage Fragments.
+ * @author anistal
+ */
+class FragmentSupervisorActor(curatorFramework: CuratorFramework) extends Actor
+with Json4sJacksonSupport with SLF4JLogging {
+
+  implicit val json4sJacksonFormats = DefaultFormats +
+    new EnumNameSerializer(StreamingContextStatusEnum) +
+    new JsoneyStringSerializer()
+
+  override def receive: Receive = {
+    case FragmentSupervisorActor_findByTypeAndName(fragmentType, name) => doDetail(fragmentType, name)
+    case FragmentSupervisorActor_create(fragment) => doCreate(fragment)
+    case FragmentSupervisorActor_deleteByTypeAndName(fragmentType, name) => doDeleteByTypeAndName(fragmentType, name)
+    case FragmentSupervisorActor_findAllByType(fragmentType) => doFindAllByType(fragmentType)
+  }
+
+  def doFindAllByType(fragmentType: String): Unit =
+    try {
+      val children = curatorFramework.getChildren.forPath( FragmentSupervisorActor.generateFragmentPath(fragmentType))
+
+      sender ! FragmentSupervisorActor_response_fragments(
+        JavaConversions.asScalaBuffer(children).toList.map(element =>
+          read[FragmentElementDto](new String(curatorFramework.getData.forPath(
+            FragmentSupervisorActor.generateFragmentPath(fragmentType) +
+              FragmentSupervisorActor.PathSeparator + element).asInstanceOf[Array[Byte]]))).toSeq)
+    } catch {
+      case exception: Exception => {
+        log.error(exception.getLocalizedMessage, exception)
+        sender ! new FragmentSupervisorActor_response_error(exception)
+      }
+    }
+
```

Comment:
This try/catch is not the scala way.

This post is very usefull
http://danielwestheide.com/blog/2012/12/26/the-neophytes-guide-to-scala-part-6-error-handling-with-try.html